plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -195,4 +214,45 @@ public static Object[][] timestampTypes()
                 {"timestamp(12)"}
         };
     }
+
+    //TODO replace TableNameDataProvider and ColumnNameDataProvider with ObjectNameDataProvider
+    // to one big single list of all special character cases, current list has additional special bracket cases,
+    // please don't forget to use this list as base
+    @DataProvider
+    public Object[][] testTableNameDataProvider()
+    {
+        return testTableNameTestData().stream()
+                .collect(toDataProvider());
+    }
+
+    private List<String> testTableNameTestData()
+    {
+        return ImmutableList.<String>builder()
+                .add("lowercase")
+                .add("UPPERCASE")
+                .add("MixedCase")
+                .add("an_underscore")
+                .add("a-hyphen-minus") // ASCII '-' is HYPHEN-MINUS in Unicode
+                .add("a space")
+                .add("atrailingspace ")
+                .add(" aleadingspace")
+                .add("a.dot")
+                .add("a,comma")
+                .add("a:colon")
+                .add("a;semicolon")
+                .add("an@at")
+                .add("a\"quote")
+                .add("an'apostrophe")
+                .add("a`backtick`")
+                .add("a/slash`")
+                .add("a\\backslash`")
+                .add("adigit0")
+                .add("0startwithdigit")
+                .add("[brackets]")
+                .add("[brack[]ets inside]")
+                .add("[brackets[ ]spaceinside]")
+                .add("single]openbrackets")
```

Comment:
that's a closing bracket, but the word says "open"

```suggestion
                .add("open[bracket")
```